plugins/autosuspend.koplugin/main.lua

```diff
@@ -177,37 +181,65 @@ function AutoSuspend:_unschedule_standby()
     end
 end
 
-function AutoSuspend:_reschedule_standby()
-    if not Device:canStandby() then return end
+function AutoSuspend:_schedule_standby()
+    -- Start the long list of conditions in which we do *NOT* want to go into standby ;).
+    if not Device:canStandby() then
+        -- NOTE: This partly duplicates what `_enabledStandby` does,
+        --       but it's here to avoid logging noise on devices that can't even standby ;).
+        return
+    end
 
-    -- We may have just disabled the feature, so unschedule before checking it.
-    self:_unschedule_standby()
+    -- Don't even schedule standby if we haven't set a proper timeout yet.
+    if not self:_enabledStandby() then
+        logger.dbg("AutoSuspend: No timeout set, no standby")
+        return
+    end
+
+    -- When we're in a state where entering suspend is undesirable, we simply postpone the check by the full delay.
+    local delay_standby = self.auto_standby_timeout_seconds
+    if NetworkMgr:isWifiOn() then
+        -- Don't enter standby if wifi is on, as this will break in fun and interesting ways (from Wi-Fi issues to kernel deadlocks).
+        logger.dbg("AutoSuspend: WiFi is on, delaying standby")
+    elseif Device.powerd:isCharging() and not Device:canPowerSaveWhileCharging() then
+        -- Don't enter standby when charging on devices where charging prevents entering low power states.
+        logger.dbg("AutoSuspend: charging, delaying standby")
+    else
+        local now_tv = UIManager:getElapsedTimeSinceBoot()
+        delay_standby = (self.last_action_tv - now_tv):tonumber() + self.auto_standby_timeout_seconds
```

Comment:
This doesn't read really well - spent a minute wondering... I'm used to read "now - last_action".
The first part is a negative number (so, a bit of gymnastic needed to get what you aimed at), and "delay_standby" sounds like a bool, "standby_delay" or "delay_before_standby" would sound better.

Reply:
That's based on the existing semantics for suspend/shutdown ;).

(That also explains why the variable name uses a suffix and not a prefix: because there's delay_suspend and delay_shutdown right next to each other).

Reply:
Also, right now, 2AM, I'm completely unable to compute how to build a delay based on now - last ;D.

Reply:
Of course you are able ! You just did :)
`standby_delay = self.auto_standby_timeout_seconds - (now_tv - self.last_action_tv):tonumber()` reads quite well.